docusaurus/src/theme/SearchBar/DocSearchModal.js

```diff
@@ -0,0 +1,214 @@
+import { useLocation } from '@docusaurus/router';
+import { Footer } from './Footer';
+import { SearchBox } from './SearchBox';
+import { Hits } from './Hits';
+
+import { useTouchEvents } from './useTouchEvents';
+import { useTrapFocus } from './useTrapFocus';
+
+import { platformMapping, folderMapping } from '../../../constants';
+
+// It doesnt make sense to use env vars for this while we are still
+// keeping the zipped file on the repo
+const algoliaClient = algoliasearch(
+  '7RY30ISS74',
+  'fd1b03de28081b5aa29dbccced4620b9'
+);
+
+const index = algoliaClient.initIndex('DOCS');
+
+export function DocSearchModal({
+  onClose = () => null,
+  initialScrollY = 0,
+  locationPlatform,
+}) {
+  const containerRef = React.useRef(null);
+  const formElementRef = React.useRef(null);
+  const dropdownRef = React.useRef(null);
+  const inputRef = React.useRef(null);
+  const modalRef = React.useRef(null);
+  const snippetLength = React.useRef(10);
+  const location = useLocation();
+  const locationQuery = useMemo(() => {
+    const { query } = url.parse(location.search || '', {
+      parseQueryString: true,
+    });
+    return query;
+  }, [location.search]);
+  const [state, setState] = React.useState({
+    query: locationQuery.query || '',
+  });
+
+  React.useEffect(() => {
+    document.body.classList.add('DocSearch--active');
+
+    return () => {
+      document.body.classList.remove('DocSearch--active');
+
+      // IE11 doesn't support `scrollTo` so we check that the method exists
+      // first.
+      window.scrollTo?.(0, initialScrollY);
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    const isMobileMediaQuery = window.matchMedia('(max-width: 750px)');
+
+    if (isMobileMediaQuery.matches) {
+      snippetLength.current = 5;
+    }
+  }, []);
+
+  React.useEffect(() => {
+    function setFullViewportHeight() {
+      if (modalRef.current) {
+        const vh = window.innerHeight * 0.01;
+        modalRef.current.style.setProperty('--docsearch-vh', `${vh}px`);
+      }
+    }
+
+    setFullViewportHeight();
+
+    window.addEventListener('resize', setFullViewportHeight);
+
+    return () => {
+      window.removeEventListener('resize', setFullViewportHeight);
+    };
+  }, []);
+
+  const autocomplete = React.useMemo(
+    () =>
+      createAutocomplete({
+        id: 'search',
+        defaultActiveItemId: 0,
+        placeholder: 'Search',
+        onStateChange({ state, setCollections }) {
+          if (!state.query && state.collections.length > 0) {
+            return setCollections([]);
+          }
+          setState(state);
+        },
+        getSources({ query }) {
+          return index
+            .search(query, {
+              filters: `parent_section_slug:chat_docs AND platforms:${platformMapping[locationPlatform]}`,
+            })
+            .then(({ hits }) => {
+              const grouped = hits.reduce((acc, hit) => {
```

Comment:
Partially what we are already doing in the old docs website. The main difference now is that we need one more level of categorization so we can have results split like that:

![image](https://user-images.githubusercontent.com/18744505/119169922-f2d65600-ba38-11eb-94c5-a681ca091a6d.png)